Address AdamISZ's feedback regarding duplicate public keys

jonasnick opened the pull request.

AdamISZ commented:
I still feel like this wording does not convey to the reader the fact that not every one of an attacker's indices can be identified in case of receiving an incorrect contribution. There's considerable finesse here as per our discussion on-list, i.e. to what extent does it matter, but clearly we're trying to be precise, maybe just something like this?:

`a signer who prevents a signing session from completing successfully by sending incorrect contributions in the session can be identified (at at least one signing index, if they are in control of more than one) and held accountable`

But obviously your call.

jonasnick commented:
I'll wait for the other authors to weigh in.

real-or-random commented:
I think the example of forgeability of  partial sigs is somewhat besides the point. When it comes to disruption and the ability to identify malicious signers ("identifiable aborts"), this property holds in MuSig2 even for maliciously created keys, i.e., whether the attacker knows a secret key or not is relevant for unforgeability but not for identifiable aborts. (See also the ROAST paper, where we make this formal for FROST; the exact same argument applies to MuSig2.)  So in the example where the attacker controls signers 4 and 5, and doesn't have the secret key of 4 but still manages to produce a contribution that passes partial verification, you could say that 4 behaves honestly with respect to identifiable aborts. It doesn't really matter that the attacker doesn't know the secret key. (The attacker could equivalently also know it and not use it.) Similarly, the attacker could make signer 4 always use the same two EC points in the nonce, or always choose the secret key to be the constant 42. Such a signer is clearly not honest in the sense that it doesn't adhere to the protocol but we simply don't care when it comes to identifiable aborts. 

So I believe one could argue that the protocol really identifies all signers which *behave disruptively in a particular session*. Of course, we can't identify all signers controlled by the attacker, because they could behave honestly in this particular session (and possibly be disruptive only in future sessions, if such sessions exist). But I totally see that this is subtle and debatable, and we shouldn't go into these details in the BIP. In the end, the protocol guarantees the following property, which is all that users of the protocol need to know, and so this is all we need to say in the BIP:

*If the signing session fails to output a valid signature, then each honest signer will identify exactly one disruptive signer, who sent incorrect contributions in the session, and return the index of the disruptive signer.*

And I believe this is in line with @AdamISZ's suggestions, (I hope) I'm just rephrasing. 

We should also mention this very useful property:
*Additionally, if the honest signers agree on the set of messages sent by all signers in the signing session, then the honest signers will identify the same disruptive signer.*



(We can replace "disruptive" by  "malicious", open to bikeshedding here.)

AdamISZ commented:
Thanks @real-or-random that's a really interesting way of looking at it.

On reflection from that, I think I can see that ultimately what I'm worried about is something that's cautioned already, expicitly, in the document, namely that partial signatures are not proofs of knowledge. As per [here](https://github.com/jonasnick/bips/blob/686327edc3c47a23455fa65a21c65ca30c988c79/bip-musig2.mediawiki#identifiying-disruptive-signers):


> Note that partial signatures are not signatures. An adversary can forge a partial signature, i.e., create a partial signature without knowing the secret key for the claimed public key[[2](https://github.com/jonasnick/bips/blob/686327edc3c47a23455fa65a21c65ca30c988c79/bip-musig2.mediawiki#cite_note-2)].

Anybody sufficiently thoughtful to 'do this right' will presumably, in any meta-protocol, take account of this fact. It could be a "gotcha" in some use case, but this BIP can't really do more than inform of that fact.

> If the signing session fails to output a valid signature, then each honest signer will identify exactly one disruptive signer, who sent incorrect contributions in the session, and return the index of the disruptive signer.

This looks good to me but: I don't get why you say 'exactly one', can't more than one signer output garbage/not follow the rules?

real-or-random commented:
> This looks good to me but: I don't get why you say 'exactly one', can't more than one signer output garbage/not follow the rules?

Ok yes, I said  "exactly one" because our algorithms currently output only one. (see also https://github.com/jonasnick/bips/pull/9 ).  But on a second thought, that's not really true... I mean sure, for example `PartialVerify` outputs only one index but that's just because this algorithm is just looking at the contributions of a single signer. So I agree, it's better to drop the "exactly". Maybe "at least one" or just "one".

jonasnick commented:
Added @real-or-random's suggestion (slightly rephrased, since PartialSigVerify doesn't return the index of the signer, that's something the application needs to handle) and moved this part to the "Identifying Disruptive Signers" section.

AdamISZ commented:
ACK https://github.com/jonasnick/bips/pull/25/commits/3aabe68599923ac80cb95a934afb72dbd9f9a05f

real-or-random commented on bip-musig2.mediawiki:
```diff
@@ -94,7 +94,12 @@ This is by design: All algorithms in this specification handle multiple signers
+In contrast, MuSig2 is designed to identify disruptive signers at signing time (see [[#identifiying-disruptive-signers|Identifiying Disruptive Signers]]).
+
+While this specification is able to deal with duplicate public keys, there are scenarios where applications may choose to abort when encountering duplicate public keys.
+We can imagine a scenario where a single entity creates a MuSig2 setup with multiple signing devices.
+In that case, duplicate public keys may not result from a malicious signing device copying a key but from accidental initialization of two devices with the same seed.
+Since this error would not be noticed by the MuSig2 protocol and reduce the security compared to the intended key setup, applications may interrupt the protocol and ask the user to investigate.
```
```suggestion
For example, we can imagine a scenario where a single entity creates a MuSig2 setup with multiple signing devices.
In that case, duplicate public keys may not result from a malicious signing device copying a public key of another signing device but from accidental initialization of two devices with the same seed.
Since MuSig2 key aggregation would accept the duplicate keys and not error out, which would in turn reduce the security compared to the intended key setup, applications may reject duplicate public keys before passing them to MuSig2 key aggregation and ask the user to investigate.
```

A few editorial suggestions, feel free to reject/change/cherry-pick. The only important point here is that I think we should mention when to reject, i.e., already before key aggregation.

real-or-random reviewed (commented):
ACK mod comments
